subsys/pm/device_runtime.c

```diff
@@ -180,12 +180,9 @@ void pm_device_runtime_enable(const struct device *dev)
 		goto unlock;
 	}
 
-	/* lazy init of PM fields */
-	if (pm->dev == NULL) {
-		pm->dev = dev;
-		pm->state = PM_DEVICE_STATE_SUSPENDED;
-		k_work_init_delayable(&pm->work, runtime_suspend_work);
-	}
+	pm->dev = dev;
+	pm->state = PM_DEVICE_STATE_SUSPENDED;
+	k_work_init_delayable(&pm->work, runtime_suspend_work);
```

Comment:
lazy init should be kept for dev and work queue, only pm->state (and usage count) should be set every time.

Reply:
Ok, I  did it because I have another upcoming change that will bind device in other place, but I can let this change to there.